packages/pwa-kit-dev/src/configs/webpack/plugins.js

```diff
@@ -58,20 +64,30 @@ export const createModuleReplacementPlugin = (projectDir) => {
         })
     })
 
+    // Return a new `webpack.NormalModuleReplacementPlugin` instance
     return new webpack.NormalModuleReplacementPlugin(/.*/, (resource) => {
-        const resolved = path.resolve(resource.context, resource.request)
+        // We only want to replace resources that are requested from the SDK
+        if (resource.context.includes('pwa-kit-react-sdk')) {
+            // Resolve the full path of the resource
+            const resolved = path.resolve(resource.context, resource.request)
 
-        const replacement = replacements.find(({path}) => resolved.match(path))
+            // Find the replacement for the resolved path from the overridables list
+            const replacement = replacements.find(({path}) => resolved.match(path))
 
-        const sdkPaths = [
-            path.join('packages', 'pwa-kit-react-sdk'),
-            path.join('node_modules', 'pwa-kit-react-sdk')
-        ]
+            if (replacement) {
+                // Check if the resource was requested from 'packages/pwa-kit-react-sdk' or 'node_modules/pwa-kit-react-sdk'
```

Comment:
I think the comment is redundant if we repeat the paths. In the future, if there are changes to the paths, the comment may be forgotten to be updated.

Sometimes it's better to comment the _why_, rather than what the the line of code actually does.

```suggestion
                // Check if the resource was requested from these paths
                // The SDK can appear in different locations depending if you're developing pwa-kit or working in your own project
```